File: 1. 단계별로_풀어보기/10~19/13. 정렬/9. 1181.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const wordArr = inputArr.slice(1)

// 정렬
wordArr.sort((a, b) => {
  if (a.length < b.length) return -1
  else if (a.length === b.length) return a.localeCompare(b)
  else return 1
})

// 출력
for (let i = 0; i < wordArr.length; i++) {
  if (i > 0 && wordArr[i - 1] === wordArr[i]) continue

  process.stdout.write(wordArr[i] + '\n')
}

File: 1. 단계별로_풀어보기/10~19/13. 정렬/9. 1181.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

wordArr = list(input().strip() for i in range(N))

wordArr.sort(key=lambda x: (len(x), x))

for i in range(N):
  if i > 0 and wordArr[i] == wordArr[i - 1]:
    continue
  output(f"{wordArr[i]}\n")